common/src/test/java/com/microsoft/identity/common/ropc/PasswordGrantTest.java

```diff
@@ -0,0 +1,165 @@
+// Copyright (c) Microsoft Corporation.
+// All rights reserved.
+//
+// This code is licensed under the MIT License.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files(the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and / or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions :
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+package com.microsoft.identity.common.ropc;
+
+import com.microsoft.identity.common.exception.ClientException;
+import com.microsoft.identity.common.internal.providers.oauth2.OAuth2Strategy;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenRequest;
+import com.microsoft.identity.common.internal.providers.oauth2.TokenResult;
+import com.microsoft.identity.common.internal.util.StringUtil;
+import com.microsoft.identity.internal.testutils.MicrosoftStsRopcTokenRequest;
+import com.microsoft.identity.internal.testutils.authorities.AADTestAuthority;
+import com.microsoft.identity.internal.testutils.labutils.Credential;
+import com.microsoft.identity.internal.testutils.labutils.Scenario;
+import com.microsoft.identity.internal.testutils.labutils.TestConfigurationQuery;
+
+import org.junit.Ignore;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.robolectric.RobolectricTestRunner;
+
+import java.io.IOException;
+import java.util.Arrays;
+import java.util.HashSet;
+import java.util.Set;
+
+import static org.junit.Assert.assertEquals;
+import static org.junit.Assert.fail;
+
+/**
+ * This class contains all tests for the password grant.
+ * All different success and failure test cases for the password grant can be added here
+ */
+@RunWith(RobolectricTestRunner.class)
+public final class PasswordGrantTest {
+
+    private static final String[] SCOPES = {"user.read", "openid", "offline_access", "profile"};
+    private static final String CLIENT_ID = "4b0db8c2-9f26-4417-8bde-3f0e3656f8e0";
```

Comment:
Just checking that you are intentionally including this client id in this PR.

Confirm?

Reply:
Yes, this is the same client id used by config for MsalTestApp